services/QuillLMS/app/controllers/application_controller.rb

```diff
@@ -147,7 +147,7 @@ def non_standard_route_redirect?(route)
 
     # Assuming that the refresh_token expires at (current_user.auth_credential.created_at  + 6 months),
     # we can reset the session whenever (Time.now > (current_user.auth_credential.created_at + 5 months))
-    return reset_session if current_user.google_id && current_user.auth_credential && Time.now > (current_user.auth_credential.created_at + 5.months)
+    return reset_session if current_user.google_id && current_user.auth_credential && (Time.now > (current_user.auth_credential.expires_at) || current_user.auth_credential.expires_at.nil?)
```

Comment:
I think you could consolidate the logic and avoid violating the Law of Demeter with:
`return reset_session if current_user.google_id && !current_user.google_authorized?`